op-node/cmd/batch_decoder/fetch/fetch.go

```diff
@@ -32,45 +34,67 @@ type TransactionWithMetadata struct {
 }
 
 type Config struct {
-	Start, End   uint64
-	ChainID      *big.Int
-	BatchInbox   common.Address
-	BatchSenders map[common.Address]struct{}
-	OutDirectory string
+	Start, End         uint64
+	ChainID            *big.Int
+	BatchInbox         common.Address
+	BatchSenders       map[common.Address]struct{}
+	OutDirectory       string
+	ConcurrentRequests uint64
 }
 
 // Batches fetches & stores all transactions sent to the batch inbox address in
 // the given block range (inclusive to exclusive).
 // The transactions & metadata are written to the out directory.
-func Batches(client *ethclient.Client, config Config) (totalValid, totalInvalid int) {
+func Batches(client *ethclient.Client, config Config) (totalValid, totalInvalid uint64) {
 	if err := os.MkdirAll(config.OutDirectory, 0750); err != nil {
 		log.Fatal(err)
 	}
-	number := new(big.Int).SetUint64(config.Start)
 	signer := types.LatestSignerForChainID(config.ChainID)
+	concurrentRequests := int(config.ConcurrentRequests)
+
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(concurrentRequests)
+
+loop:
 	for i := config.Start; i < config.End; i++ {
-		valid, invalid := fetchBatchesPerBlock(client, number, signer, config)
-		totalValid += valid
-		totalInvalid += invalid
-		number = number.Add(number, common.Big1)
+		select {
+		case <-ctx.Done():
+			break loop
+		default:
+		}
```

Comment:
```suggestion
		if err := ctx.Err(); err != nil {
			break 
		}
```
No need to use a select statement here, checking ctx.Err() does the same thing. (optional, it just removes the need for the `break loop` statement).